createdisk.sh

```diff
@@ -11,19 +11,12 @@ OC=${OC:-oc}
 DEVELOPER_USER_PASS='developer:$2y$05$paX6Xc9AiLa6VT7qr2VvB.Qi.GJsaqS80TR3Kb78FEIlIL0YyBuyS'
 
 function get_git_tag {
-    GIT_TAG=$(git describe --exact-match --tags HEAD) || GIT_TAG=
-
-    # Based on code from git-version-gen
-    # Don't declare a version "dirty" merely because a time stamp has changed
-    git update-index --refresh > /dev/null 2>&1
-
-    dirty=`sh -c 'git diff-index --name-only HEAD' 2>/dev/null` || dirty=
-    case "$dirty" in
-        '') ;;
-        *) # Don't build an 'official' version if git tree is dirty
-            GIT_TAG=
-    esac
-    # end of git-version-gen code
+    BRANCH=$(git rev-parse --abbrev-ref HEAD)
+    if [[ "$BRANCH" == "master" ]]; then
+      GIT_TAG=
+      return
+    fi
+    GIT_TAG=$(git describe --exact-match HEAD 2>/dev/null) || GIT_TAG=
```

Comment:
So this means the git tag versioning code will only work if we create branches for releases? Is this what we are going to do? If latest git master commit is tagged, then `git rev-parse --abbrev-ref HEAD` will return `master`, and we'll never run the 'git describe` code. This means we went from "hard to create snapshots" to "hard to create releases" imo.

Reply:
@cfergeau right, even if the master branch have commit to a tag then also if a user manually checkout to a released tag it will not able to build the tagged version.